src/Cms/PageActions.php

```diff
@@ -434,18 +434,38 @@ public function copy(array $options = [])
 
 		$copy = $parentModel->clone()->findPageOrDraft($slug);
 
-		// remove all translated slugs
 		if ($this->kirby()->multilang() === true) {
+			// remove all translated slugs
 			foreach ($this->kirby()->languages() as $language) {
-				if ($language->isDefault() === false && $copy->translation($language)->exists() === true) {
+				if (
+					Uuids::enabled() === true &&
+					$language->isDefault() === true
+				) {
+					// overwrite with new UUID for the page and files
+					// for default language (remove old, add new)
+					$copy = $copy->save(['uuid' => Uuid::generate()], $language->code());
+
+					if ($files !== false) {
+						foreach ($copy->files() as $file) {
+							$file->save(['uuid' => Uuid::generate()], $language->code());
```

Comment:
I'm not sure that is there anything we need to do in the cached data/objects for updated files?

Reply:
Which caches do you mean in particular? I could well be wrong, but I think the object won't yet be cached anywhere because it was newly created (it's a new copy after all). If we *do* have a caching issue here, it will likely also affect the copied page object.

Reply:
You're right. A fresh copy and the `::updateParentCollections()` method is working last. Then there is no problem 👍

Reply:
Sometimes I get confused because of this immutable. I've one more question about that; 

````php
foreach ($copy->files() as $file) {
  $file->save(['uuid' => Uuid::generate()], $language->code());
}
````

When the `$file->save()` method is called, the file objects in `$copy->files()` are also updated, right?
So the `$copy` object now has file objects with new UUIDs, right?

Reply:
I really hope so. At least it should be like that if our code is correct. But to be honest, I would need to test it myself.